packages/block-editor/src/store/reducer.js

```diff
@@ -1737,11 +1737,17 @@ export function highlightedBlock( state, action ) {
  */
 export function lastBlockInserted( state = {}, action ) {
 	switch ( action.type ) {
-		case 'ADD_LAST_BLOCK_INSERTED':
-			return { ...state, clientId: action.clientId };
+		case 'INSERT_BLOCKS':
+			if ( ! action.updateSelection || ! action.blocks.length ) {
+				return state;
+			}
 
-		case 'CLEAR_LAST_BLOCK_INSERTED':
-			return {};
+			const clientId = action.blocks[ 0 ].clientId;
+			const source = action.meta?.source;
+
+			return { clientId, source };
+		case 'RESET_BLOCKS':
+			return null;
```

Comment:
Quick question here, in my implementation I passed an empty object to clear `lastBlockInserted` and here you are passing null. Is there a standard for this?

Reply:
Oh good catch! Here we should return an empty object because that's the default value for this state. I'm going to push the change asap.

Reply:
I fixed this issue in this [commit](https://github.com/WordPress/gutenberg/pull/30238/commits/1164a3338fa9347a52bffcc34283d0986daf2e2c).